rero_mef/api.py

```diff
@@ -144,6 +144,13 @@ def create_or_update(cls, data, id_=None, delete_pid=True, dbcommit=False,
         if agent_record := cls.get_record_by_pid(pid):
             # record exist
             data = add_schema(data, agent_record.provider.pid_type)
+            # save the records old data if the new one is empty
+            for to_save in [
+                    'pid', '$schema', 'identifier', 'identifiedBy',
+                    'authorized_access_point', 'bf:Agent', 'relation_pid',
+                    'deleted']:
+                if not data.get(to_save) and agent_record.get(to_save):
+                    data[to_save] = agent_record[to_save]
```

Comment:
suggestion : 
 * filter the `dict` to keep only required keys/values
 * use dict merge to add/override values
 ```python
copy_fields = ['pid', '$schema', 'identifier', 'identifiedBy', ...]
original_data = {k: v for k, v in agent_record.items if k in copy_fields}
data = {**original_data, **data}  # dict merging, `original_data` values will be override by `data` values
```